std/concurrency.d

```diff
@@ -656,6 +634,36 @@ body
     thisInfo.ident.mbox.get( ops );
 }
 
+///
+@system unittest
+{
+    import std.variant;
+    __gshared string received;
+
+    static void spawnedFunction()
+    {
+        receive(
+            (int i) {
+                received = "Received an int.";
+                ownerTid.send(true); // signal success
```

Comment:
If you want to check that the functions are called, I suggest something along the lines of:
```D
auto process = ()
{
    receive(
        (int i) { ownerTid.send(1); },
        (double f) { ownerTid.send(2); },
        (Variant v) { ownerTid.send(3); }
    );
};

{
    auto tid = spawn(process);
    send(tid, 42);
    assert(receiveOnly!int == 1);
}

{
    auto tid = spawn(process);
    send(tid, 3.14);
    assert(receiveOnly!int == 2);
}

{
    auto tid = spawn(process);
    send(tid, "something else");
    assert(receiveOnly!int == 3);
}
```

Otherwise, it would be be better to keep the old examples.